netmiko/dell/dell_sonic_ssh.py

```diff
@@ -0,0 +1,54 @@
+"""Netimko driver for Dell EMC PowerSwitch platforms running Dell EMC Open Networking Powered by SONiC"""
+import time
+from netmiko.cisco_base_connection import CiscoSSHConnection
+from netmiko import log
+
+class DellSonicSSH(CiscoSSHConnection):
+    """driver for Dell EMC PowerSwitch platforms running Dell EMC Open Networking Powered by SONiC"""
+
+    def session_preparation(self):
+        """Prepare the session after the connection has been established."""
+        self._test_channel_read(pattern=r"[>$#]")                
+        # Clear the read buffer
+        time.sleep(0.3 * self.global_delay_factor)
+        self.clear_buffer()
+        self._enter_shell()
+        self.set_base_prompt(alt_prompt_terminator="$")
+
+    def check_config_mode1(self, check_string="#", pattern=""):
+        log.debug(f"pattern: {pattern}")
+        self.write_channel(self.RETURN)
+        output = self.read_until_pattern(pattern=pattern)
+        log.debug(f"check_config_mode: {repr(output)}")
+        output = output.replace("(s1)", "")
+        output = output.replace("(s2)", "")
+        log.debug(f"check_config_mode: {repr(output)}")
+        return check_string in output
+
+    def disable_paging(self, command="terminal length 0", delay_factor=1):
+        """Disable paging sets the command: terminal length 0
+
+        :param command: Device command to disable pagination of output
+        :type command: str
+        :param delay_factor: See __init__: global_delay_factor
+        :type delay_factor: int
+        """
+        return self.send_command(command, expect_string=r"[\#]")
```

Comment:
Why do you need a custom `disable_paging`? Can you use the existing one in the parent class(es)?